python/ray/rllib/models/action_dist.py

```diff
@@ -4,8 +4,13 @@
 
 import tensorflow as tf
 import numpy as np
+import distutils.version
+
 from ray.rllib.utils.reshaper import Reshaper
 
+use_tf150_api = (distutils.version.LooseVersion(tf.VERSION) >=
```

Comment:
```
autodoc: failed to import module 'ray.rllib.agents'; the following exception was raised:
Traceback (most recent call last):
  File "/home/travis/miniconda/lib/python3.6/site-packages/sphinx/ext/autodoc/importer.py", line 152, in import_module
    __import__(modname)
  File "/home/travis/build/ray-project/ray/python/ray/rllib/__init__.py", line 9, in <module>
    from ray.rllib.evaluation.policy_graph import PolicyGraph
  File "/home/travis/build/ray-project/ray/python/ray/rllib/evaluation/__init__.py", line 1, in <module>
    from ray.rllib.evaluation.policy_evaluator import PolicyEvaluator
  File "/home/travis/build/ray-project/ray/python/ray/rllib/evaluation/policy_evaluator.py", line 10, in <module>
    from ray.rllib.models import ModelCatalog
  File "/home/travis/build/ray-project/ray/python/ray/rllib/models/__init__.py", line 1, in <module>
    from ray.rllib.models.catalog import ModelCatalog
  File "/home/travis/build/ray-project/ray/python/ray/rllib/models/catalog.py", line 13, in <module>
    from ray.rllib.models.action_dist import (
  File "/home/travis/build/ray-project/ray/python/ray/rllib/models/action_dist.py", line 11, in <module>
    use_tf150_api = (distutils.version.LooseVersion(tf.VERSION) >=
  File "/home/travis/miniconda/lib/python3.6/distutils/version.py", line 304, in __init__
    self.parse(vstring)
  File "/home/travis/miniconda/lib/python3.6/distutils/version.py", line 312, in parse
    components = [x for x in self.component_re.split(vstring)
TypeError: expected string or bytes-like object

```

Reply:
It looks like this happened during
```


travis_time:end:00ea8427:start=1533650555964759398,finish=1533650559705240384,duration=3740480986
[0K
[31;1mThe command "sphinx-build -W -b html -d _build/doctrees source _build/html" exited with 2.[0m
```

Reply:
The issue is probably that we mock the tensorflow module when building the documentation (so that it doesn't need to be installed).

https://github.com/ray-project/ray/blob/64053278aafd565b74f7b015d1513b40d94fe24d/doc/source/conf.py#L21-L29

Not sure what the standard solution is, but it might suffice to add `sys.modules["tensorflow"].VERSION = ""` after https://github.com/ray-project/ray/blob/64053278aafd565b74f7b015d1513b40d94fe24d/doc/source/conf.py#L56-L57

It should be possible to reproduce this error by uninstalling tensorflow and then rebuilding the documentation.

Reply:
hi robertnishihara, should I change conf.py according to your example and push again?

Reply:
Can you try building the documentation locally (without TF installed) and see if you can reproduce the error, and then see if the change fixes it? If not, we can try other fixes.

Reply:
I confronted this problem: 
![image](https://user-images.githubusercontent.com/6416895/43815777-e5b2ca14-9b03-11e8-9e23-da8ee669d463.png)

I found that you have added: sys.path.insert(0, os.path.abspath("../../python/"))

So, I cannot understand why sphinx failed to find this module. Could you give me a hand?

Reply:
I just commented out this module and then `make html` can be executed.

The problem can be reproduced:
![image](https://user-images.githubusercontent.com/6416895/43816167-fcfd37a2-9b05-11e8-8b07-56fe31e91d70.png)

If added `sys.modules["tensorflow"].VERSION = ""` after the for loop, I got:
![image](https://user-images.githubusercontent.com/6416895/43816265-5c8dd49c-9b06-11e8-9736-4589f2f300f6.png)

Any idea?

Reply:
I gave "1.4.1" instead of an empty string "" to `sys.modules["tensorflow"].VERSION`, then the `make html` succeed without those warnings (actually errors they are). How should I do now? Maybe add `sys.modules["tensorflow"].VERSION = "1.4.1"` in conf.py and commit again?

Reply:
I'd suggest using a more obviously fake version like "9.9.9" and then adding an elaborate comment explaining why this is needed. Then I think this is an acceptable workaround.

Reply:
Hi robertnishihara, the CI failure info seems vague to me, could you give me some tips on this CI failure?